exec/cnex/cell.c

```diff
@@ -65,15 +61,26 @@ Cell *cell_fromString(const char *s, int64_t length)
     Cell *x = cell_newCell();
     x->type = cString;
     x->string = string_newString();
+    x->string->length = length;
 
-    if (length == -1) {
-        x->string->length = strlen(s);
-    } else {
-        x->string->length = length;
+    x->string->data = malloc(x->string->length);
+    if (x->string->data == NULL) {
+        fatal_error("Failed to allocate memory for new cell string.");
     }
+    memcpy(x->string->data, s, x->string->length);
+    return x;
+}
+
+Cell *cell_fromCString(const char *s)
```

Comment:
Changed this function to cell_fromTerminatedString() instead.  I didn't like the fromStringLength().  It didn't read well.  Hopefully this will be ok.  Also added NUL termination to resultant string.